src/thunderbots/software/radio_communication/mrf/dongle.cpp

```diff
@@ -0,0 +1,610 @@
+#include "dongle.h"
+
+#include <sigc++/bind.h>
+#include <sigc++/functors/mem_fun.h>
+#include <sigc++/reference_wrapper.h>
+#include <unistd.h>
+
+#include <algorithm>
+#include <bitset>
+#include <cassert>
+#include <chrono>
+#include <cstdlib>
+#include <cstring>
+#include <exception>
+#include <iostream>
+#include <stdexcept>
+#include <string>
+#include <unordered_map>
+
+#include "constants.h"
+#include "radio_communication/visitor/mrf_primitive_visitor.h"
+#include "util/logger/init.h"
+
+namespace
+{
+    struct RadioConfig
+    {
+        uint8_t channel;
+        int symbol_rate;
+        uint16_t pan;
+    };
+
+    const RadioConfig DEFAULT_CONFIGS[4] = {
+        {25U, 250, 0x1846U},
+        {25U, 250, 0x1847U},
+        {25U, 250, 0x1848U},
+        {25U, 250, 0x1849U},
+    };
+
+    const unsigned int ANNUNCIATOR_BEEP_LENGTH = 750;
```

Comment:
What are the units for this?

Reply:
Most likely milliseconds.

Reply:
Empirically that seems correct. Adding units in the variable name would be great.